runtime/src/main/java/io/quarkus/hazelcast/client/runtime/graal/Target_AbstractInvocationFuture.java

```diff
@@ -15,19 +17,30 @@
             return (T) exceptionClass.getConstructor(String.class, Throwable.class)
               .newInstance(cause.getMessage(), cause);
         } catch (Throwable e) {
+            try {
+                return (T) exceptionClass.getConstructor(Throwable.class).newInstance(cause);
+            } catch (Throwable e2) {
+                try {
+                    T result = (T) exceptionClass.getConstructor(String.class).newInstance(cause.getMessage());
+                    result.initCause(cause);
+                    return result;
+                } catch (Throwable e3) {
```

Comment:
🤯🤯🤯

Reply:
Yup, it's responsible for restoring and propagating stacktraces for proxied remote calls